### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by prateekgogia**
Issue #, if available:
https://github.com/awslabs/kubernetes-iteration-toolkit/issues/100
Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,31 @@
...
 
+[KIT](https://github.com/awslabs/kubernetes-iteration-toolkit) is a set of decoupled tools designed to accelerate the development of Kubernetes through testing. It combines a variety of open source projects to define an opinionated way to rapidly configure and test Kubernetes components on AWS.
 
-It consists of the following components at the moment.
+## Why did we build KIT?
 
-* [Operator](operator/README.md)
-* [KIT-environment](substrate/README.md)
-* [TestBed](testbed/README.md)
+The EKS Scalability team is responsible for improving performance across the Kubernetes stack. We started our journey by manually running tests against modified EKS dev clusters. This helped us to identify some bottlenecks, but results were difficult to demonstrate and reproduce. We wanted to increase the velocity of our discoveries, as well as our confidence in our results. We set out to build automation to help us configure cluster components, execute well known test workloads, and analyze the results. This evolved into KIT, and we’re ready to share it to help accelerate testing in other teams.
```
Comment (natherz97) [inline README.md:9]: I'm not sure if we should provide this section in the public repo, or we might want to consider removing the section talking about the difficulty of testing on EKS dev clusters.

File: README.md [hunk 2]
```diff
@@ -1,10 +1,31 @@
...
+
+## What can I do with KIT?
+
+KIT can help you run scale tests against a KIT cluster or an EKS cluster, collect logs and metrics from the cluster control plane and nodes to help analyze the performance for Kubernetes cluster. KIT comes with a set of tools like Karpenter, ELB controller, Prometheus, Grafana and Tekton etc. installed and configured to manage cluster lifecycle, run tests and collect results.
```
Comment (natherz97) [inline README.md:13]: Nit: can you update the sentence to be "for a Kubernetes cluster" and can you be consistent with capitalization on things like using KIT cluster vs. KIT Cluster?

File: README.md [hunk 3]
```diff
@@ -1,10 +1,31 @@
...
+
+## What are KIT Environments?
+
+KIT Environments provide an opinionated testing environment with support for test workflow execution, analysis, and observability. Developers can use `kitctl` cli to create a personal or shared testing environment for oneshot or periodic tests. KIT Environments are EKS Clusters that come preinstalled with a suite of Kubernetes operators that enable the execution, analysis, and persistence.
```
Comment (natherz97) [inline README.md:17]: Should we say that KIT environments are EKS clusters given kitctl will not give you an EKS cluster?

File: docs/how-to-use-kit.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
+# How to use KIT for Kubernetes testing
+
+[Kubernetes-Iteration-Toolkit](https://github.com/awslabs/kubernetes-iteration-toolkit) is a toolkit that help create a testing environment and tools to manage the lifecycle of Kubernetes clusters (EKS or vanilla Kubernetes), run tests against new or existing clusters and collect metrics & logs from the control plane for analysis.
```
Comment (natherz97) [inline docs/how-to-use-kit.md:3]: Nit: can you update the sentence to be "is a toolkit that creates a"

File: docs/how-to-use-kit.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+kitctl bootstrap kitctl-$(whoami) # Optional environment name
+```
+
+Note: Make sure you are logged into the AWS account and have admin privilege, and the environment name is unique to you as KIT uses it create an S3 bucket in your AWS account.
+
+* Bootstrap takes about 5 minutes,  as part of this bootstrap it configures the AWS infra such as VPC, security groups, subnets, route tables etc. and a management Kubernetes cluster. At the end of the bootstrap command it prints an `export KUBECONFIG` command, run this command to access the management cluster using kubectl.
+
+Note: This environment we just created comes pre-installed with Tekton for running the tests and a Prometheus stack for monitoring.
+
+* To access the Tekton dashboard `kubectl port-forward svc/tekton-dashboard -n tekton-pipelines 9097:9097` and you will see the templates for tasks and pipelines in Tekton. If you need more info about Tekton terminology <refer here> TODO
```
Comment (natherz97) [inline docs/how-to-use-kit.md:37]: Can we remove the TODOs?

File: docs/how-to-use-kit.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+* To access the Grafana dashboard `kubectl port-forward svc/kube-prometheus-stack-grafana -n monitoring 8080:80` <Add more details about the dashboards> TODO
+
+## Running tests
+
+At this point a user can run tests against either an EKS cluster or vanilla Kubernetes cluster provisioned by KIT operator called [guest clusters](https://quip-amazon.com/xCtbA6C6X7dy/How-to-use-KIT-for-Kubernetes-testing#temp:C:VfOa1f3b6a5d7de51afb05d574ef) running in the environment. 
+Difference between testing against the two cluster types-
+
+* EKS cluster doesn’t allow changing any flags or control plane images, however, if you want to make any flags changes or want to run custom docker images for API server or any other Kubernetes control plane component use guest cluster for testing
+* Metrics that we collect from EKS clusters are limited to API server metrics and CW metrics, we don’t get the node level metrics from EKS control plane nodes. In guest clusters, we are able to collect API server, etcd, scheduler, KCM and node level metrics.
```
Comment (natherz97) [inline docs/how-to-use-kit.md:46]: Should we mention that CW metrics are available given only internal teams will have these?

File: docs/how-to-use-kit.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+```
+# cd github.com/awslabs/kubernetes-iteration-toolkit/tests
+# kubectl apply -n tekton-pipelines -f pipelines -R
+# kubectl apply -n tekton-pipelines -f tasks -R
+```
+
+* Access the Tekton dashboard on [http://localhost:9097](http://localhost:9097/), make sure you have run port-forward command mentioned above.
+* Under the tasks tab look for the pre-loaded tasks that can be run.
+* Under pipelines tab are some sample pipelines flow which help create logic regarding run multiple tasks for an end-to-end test flow. Example pipeline combines the following tasks-
```
Comment (natherz97) [inline docs/how-to-use-kit.md:62]: Nit: can you say "for running multiple tasks"

File: docs/how-to-use-kit.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+kubectl label configmap master-dashboard -n monitoring  grafana_dashboard=1
+```
+>Note: we are [tracking the task](https://github.com/awslabs/kubernetes-iteration-toolkit/issues/207) on automating this step as well where the Grafana dashboards are pre-loaded in Grafana
+
+Accessing logs for kube components-
+[WIP](https://github.com/awslabs/kubernetes-iteration-toolkit/issues/106)
+
+### Use-cases supported with KIT
+
+* As a Kubernetes developer/user, create x number of objects(pods, secrets, config maps) in a cluster. As part of the validation, I want to
```
Comment (natherz97) [inline docs/how-to-use-kit.md:91]: Could this section be a little more concise? For example I don't think we need to prefix each bullet with "As a Kubernetes developer/user."

File: docs/how-to-use-kit.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+* As a Kubernetes developer/user, In an EKS cluster, I want to run an e2e scalability test to create x number of different objects and y number of nodes 
+    * Check the SLO and latency
+* As a Kubernetes developer/user, I have a custom API server/Scheduler image with some changes and want to test these change in a Kubernetes cluster, by creating x number of objects
+* As a Kubernetes developer/user, I want to run etcd or master components on specific EC2 instance types and run scale tests
+* As a Kubernetes developer/user, in an existing EKS cluster can I run a load test, get all the metrics, resource usage and validation tests
+
+
+## Key Terms
+
+**KIT/clusters or guest clusters -** These are rapid prototyping vanilla Kubernetes clusters provisioned using using EKS-distro images. They take about 3-4 minutes to provision on ec2 nodes and less than 30 seconds to update their configurations. These clusters are provisioned by KIT-operator running in the KIT environment
```
Comment (natherz97) [inline docs/how-to-use-kit.md:105]: Nit: should this be "KIT clusters or guest clusters"?

File: README.md [hunk 10]
```diff
@@ -1,10 +1,31 @@
-### What is Kubernetes Iteration Toolkit?
```
Comment (natherz97) [inline README.md:1]: Thanks for being so detailed!

File: docs/how-to-use-kit.md (hunk 9 above)
Comment (prateekgogia) [reply docs/how-to-use-kit.md:105]: We have been using these interchangeably, so want to make sure both are covered.

Comment (liko9) [reply docs/how-to-use-kit.md:105]: Line 41 references quip which is AWS internal only ;)

**Review by natherz97 — APPROVED** [review]

**Pull request merged**